null_safety_examples/misc/lib/effective_dart/design_good.dart

```diff
@@ -272,6 +310,33 @@ void miscDeclAnalyzedButNotTested() {
   };
 }
 
+//----------------------------------------------------------------------------
+
+// #docregion dont-type-init-formals
+class Point1 {
+  double x, y;
+  Point1(this.x, this.y);
+}
+// #enddocregion dont-type-init-formals
+
+//----------------------------------------------------------------------------
+
+// #docregion inferred-type-args
+class Downloader {
+  final Completer<String> response = Completer();
+}
+// #enddocregion inferred-type-args
+
+//----------------------------------------------------------------------------
+
+// #docregion redundant
+class Downloader1 {
+  final Completer<String> response = Completer();
```

Comment:
Following "DO type annotate fields and top-level variables if the type isn’t obvious." I would expect this to say:
```dart
class Downloader1 {
  final response = Completer<String>();
}
```
Line 326 is carving out an optional way to restate this but I think we should reinforce here that types should not be duplicated, if possible.

Reply:
We don't consider `Foo foo = Foo()` to be repeating the type because the thing on the right is a constructor call, not a type annotation. And for fields and top-level variables, I think there is a general preference (but not a hard rule) towards annotating the variable instead of putting type arguments on the invocation.

Reply:
I'm gonna mark this resolved because I think the guidelines here say what we intend to say (and are in line with what I think has long been the recommended guidance), but I'm still open to this discussion if you want to.